tests/test_crawlera.py

```diff
@@ -35,6 +35,13 @@ class CrawleraMiddlewareTestCase(TestCase):
     def setUp(self):
         self.spider = Spider('foo')
         self.settings = {'CRAWLERA_APIKEY': 'apikey'}
+        Response_init_orig = Response.__init__
+
+        def Response_init_new(self, *args, **kwargs):
+            assert not kwargs.get('request'), 'response objects at this stage shall not be pinned'
+            return Response_init_orig(self, *args, **kwargs)
+
+        Response.__init__ = Response_init_new
```

Comment:
@starrify I think this could be dangerous, we don't really need to make sure this is true for all tests here, I think you should only mock it in one test.

Any reason to believe ALL responses being processed by the middleware's` process_response` are not associated with a request?

Reply:
I see your point. Specially given @elacuesta’s https://github.com/scrapy/scrapy/pull/4632.

Maybe we could read `meta` from `response` and only fall back to `request.meta` if the former fails. And test both scenarios.

Reply:
functionality is ok I think, I just think overriding the `Response` object for all tests is dangerous, here we are assuming all responses should come without a request associated for ALL of our tests, which is something I don't think we need, this could potentially have repercussions on future tests/functionality (tests are now working with and without this override btw)

Reply:
I'll merge this PR and create a new issue to handle both cases fo safety